partiql-spi/src/main/kotlin/org/partiql/spi/value/ion/IonDatumReader.kt

```diff
@@ -66,36 +67,64 @@ internal class IonDatumReader internal constructor(
     /**
      * Read without any explicit PartiQL type information.
      */
-    private fun read(): Datum = when (reader.type) {
-        IonType.NULL -> Datum.nullValue()
-        IonType.BOOL -> bool()
-        IonType.INT -> bigint()
-        IonType.FLOAT -> double()
-        IonType.DECIMAL -> decimal0()
-        IonType.TIMESTAMP -> TODO("timestamp without annotation")
-        IonType.STRING -> varchar0()
-        IonType.CLOB -> clob0()
-        IonType.BLOB -> clob0()
-        IonType.LIST -> array()
-        IonType.STRUCT -> struct()
-        IonType.SYMBOL -> missing()
-        IonType.SEXP -> {
-            reader.stepIn()
-            if (reader.next() == null) {
-                throw IonDatumException("expected type, was null", null, span())
-            }
-            val method = method()
-            if (reader.next() == null) {
-                throw IonDatumException("expected value, was null", null, span())
-            }
-            val value = method()
-            if (reader.next() != null) {
-                throw IonDatumException("expected end of s-expression pair", null, span())
+    private fun read(): Datum {
+        if (reader.isNullValue) {
+            return Datum.nullValue(reader.type.toPTypeDefault())
+        }
+        return when (reader.type) {
+            IonType.NULL -> Datum.nullValue()
+            IonType.BOOL -> bool()
+            IonType.INT -> bigint()
+            IonType.FLOAT -> double()
+            IonType.DECIMAL -> decimal0()
+            IonType.TIMESTAMP -> TODO("timestamp without annotation")
+            IonType.STRING -> varchar0()
+            IonType.CLOB -> clob0()
+            IonType.BLOB -> clob0()
+            IonType.LIST -> array()
+            IonType.STRUCT -> struct()
+            IonType.SYMBOL -> missing()
```

Comment:
I think Ion symbols should be considered strings when reading/converting to PartiQL values.

> symbol: provides efficient encoding for frequently occurring strings 

from: https://amazon-ion.github.io/ion-docs/

> In binary Ion, common text tokens such as struct field names are automatically stored in a symbol table. This allows these tokens to be efficiently encoded as table offsets instead of repeated copies of the same text. As a further space optimization, symbol tables can be pre-shared between producer and consumer so that only the table name and version are included in the payload, eliminating the overhead involved with repeatedly defining the same symbols across multiple pieces of Ion data.

from: https://amazon-ion.github.io/ion-docs/guides/why.html#rich-type-system

Cf. https://amazon-ion.github.io/ion-docs/docs/symbols.html